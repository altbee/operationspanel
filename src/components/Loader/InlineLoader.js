import React from "react";
import PropTypes from "prop-types";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";

const InlineLoader = props => {
  const { visible, className } = props;
  return (
    <span className={!visible ? "d-none" : ""}>
      <FontAwesomeIcon
        icon="circle-notch"
        className={`text-primary ${className}`}
        spin
      />
    </span>
  );
};

InlineLoader.propTypes = {
  visible: PropTypes.bool,
  className: PropTypes.string
};

InlineLoader.defaultProps = {
  visible: false,
  className: ""
};

export default InlineLoader;
