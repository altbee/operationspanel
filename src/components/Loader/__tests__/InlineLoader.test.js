import React from "react";
import { configure, shallow } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import InlineLoader from "../InlineLoader";

configure({ adapter: new Adapter() });

describe("<InlineLoader />", () => {
  const props = {
    visible: true,
    className: "test-classname"
  };

  it("accepts props and renders", () => {
    const rendered = shallow(<InlineLoader {...props} />);
    expect(rendered.find("span").hasClass("")).toEqual(true);
    expect(
      rendered.find(`span > FontAwesomeIcon.${props.className}`).exists()
    ).toBe(true);
  });

  it("loads with default props", () => {
    const rendered = shallow(<InlineLoader />);
    expect(rendered.find("span").hasClass("d-none")).toEqual(true);
    expect(rendered.find(`span > i.${props.className}`).exists()).toBe(false);
  });

  it("matches the snapshot", () => {
    const tree = shallow(<InlineLoader {...props} />);
    expect(tree).toMatchSnapshot();
  });
});
