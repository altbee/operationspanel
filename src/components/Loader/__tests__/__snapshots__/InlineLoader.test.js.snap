// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<InlineLoader /> matches the snapshot 1`] = `
<span
  className=""
>
  <FontAwesomeIcon
    border={false}
    className="text-primary test-classname"
    fixedWidth={false}
    flip={null}
    icon="circle-notch"
    inverse={false}
    listItem={false}
    mask={null}
    pull={null}
    pulse={false}
    rotation={null}
    size={null}
    spin={true}
    symbol={false}
    title=""
    transform={null}
  />
</span>
`;
